pkg/container-utils/cri/cri.go

```diff
@@ -153,6 +159,17 @@ func parseContainerDetailsData(runtimeName string, containerStatus *pb.Container
 		},
 	}
 
+	// Fill K8S information.
+	if podName, ok := containerStatus.Labels[containerLabelK8sPodName]; ok {
+		containerDetailsData.PodName = podName
+	}
+	if podNamespace, ok := containerStatus.Labels[containerLabelK8sPodNamespace]; ok {
+		containerDetailsData.PodNamespace = podNamespace
+	}
+	if podUID, ok := containerStatus.Labels[containerLabelK8sPodUID]; ok {
+		containerDetailsData.PodUID = podUID
+	}
```

Comment:
Isn't this information already available in the response of `ListContainers`? If so, please add it there instead.

Reply:
Do you mean that I move the pod fields from `ContainerDetailsData` to `ContainerData` and then provide them in both list and details flows? 
Parsing can't be exact same code as runtime client structures differ between the flows.

Reply:
> Do you mean that I move the pod fields from ContainerDetailsData to ContainerData and then provide them in both list and details flows?
>
Exactly. I would like to have this information when also calling `GetContainer` and `GetContainers` as we have a use-case on Local-Gadget when calling `GetContainerDetails` will fail because the container is not yet running. 

> Parsing can't be exact same code as runtime client structures differ between the flows.
> 
Mmmm, what about this:

```golang
func EnrichWithK8sMetadata(container *runtimeclient.ContainerData, labels map[string]string) {
	if podName, ok := labels[containerLabelK8sPodName]; ok {
		container.PodName = podName
	}
	if podNamespace, ok := labels[containerLabelK8sPodNamespace]; ok {
		container.PodNamespace = podNamespace
	}
	if podUID, ok := labels[containerLabelK8sPodUID]; ok {
		container.PodUID = podUID
	}
}
```

Then, from `parseContainerDetailsData`:

```suggestion
	// Fill K8S information.
	enrichWithK8sMetadata(&containerDetailsData.ContainerData, containerStatus.Labels)
```

And from `CRIContainerToContainerData`:

```golang
func CRIContainerToContainerData(runtimeName string, container *pb.Container) *runtimeclient.ContainerData {
	containerData := &runtimeclient.ContainerData{
		ID:      container.Id,
		Name:    strings.TrimPrefix(container.GetMetadata().Name, "/"),
		State:   containerStatusStateToRuntimeClientState(container.GetState()),
		Runtime: runtimeName,
	}

	enrichWithK8sMetadata(containerData, container.Labels)

	return containerData
}
```

BTW, `EnrichWithK8sMetadata` could be defined in `runtime-client/interface.go` so that it is used by CRI and Docker code. WDYT?